wallet/src/main/java/com/android/identity_credential/wallet/QrEngagementViewModel.kt

```diff
@@ -0,0 +1,99 @@
+package com.android.identity_credential.wallet
+
+import android.app.Application
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.android.identity.android.mdoc.engagement.QrEngagementHelper
+import com.android.identity.android.mdoc.transport.DataTransport
+import com.android.identity.android.mdoc.transport.DataTransportOptions
+import com.android.identity.internal.Util
+import com.android.identity.mdoc.connectionmethod.ConnectionMethod
+import com.android.identity.mdoc.connectionmethod.ConnectionMethodBle
+import com.android.identity.securearea.SecureArea
+import com.android.identity.util.Logger
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.launch
+import java.security.KeyPair
+import java.util.UUID
+
+class QrEngagementViewModel(val context: Application) : AndroidViewModel(context)  {
+    companion object {
+        private const val TAG = "MainViewModel"
+    }
+
+    init {
+        Logger.i(TAG, "new mainViewModel")
+    }
+    var qrEngagementHelper: QrEngagementHelper? = null
```

Comment:
`qrEngagementHelper` should be private. All you really need is the QR code so suggest to just add a property with a getter like e.g. `get() = qrEngagementHelper.deviceEngagementUriEncoded` ... also, just call that property `qrCode`, no need for the Compose bits to know about the fact that it contains DeviceEngagement according to 18013-5.